src/Compilers/VisualBasic/Portable/Binding/Binder_Lookup.vb

```diff
@@ -1490,30 +1532,35 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         leaveEventsOnly = isEventsOnlySpecified
                     End If
 
-                    If lookupResult.IsGood AndAlso currentResult.IsGood Then
+                    If lookupResult IsNot Nothing AndAlso lookupResult.IsGood AndAlso currentResult.IsGood Then
                         ' We have _another_ viable result while lookupResult is already viable. Use special interface merging rules.
                         MergeInterfaceLookupResults(lookupResult, currentResult, basesBeingResolved, leaveEventsOnly, useSiteDiagnostics)
                     Else
                         If currentResult.IsGood AndAlso leaveEventsOnly.HasValue Then
                             FilterSymbolsInLookupResult(currentResult, SymbolKind.Event, leaveInsteadOfRemoving:=leaveEventsOnly.Value)
                         End If
-                        lookupResult.MergePrioritized(currentResult)
+
+                        If lookupResult Is Nothing AndAlso currentResult.Kind <> LookupResultKind.Empty Then
+                            lookupResult = LookupResult.GetInstance()
+                        End If
+
+                        lookupResult?.MergePrioritized(currentResult)
                     End If
                     currentResult.Clear()
 
                 Loop While lookIn.Count <> 0
 
                 currentResult.Free()
 
-                If methodsOnly AndAlso lookupResult.IsGood Then
+                If methodsOnly AndAlso lookupResult IsNot Nothing AndAlso lookupResult.IsGood Then
```

Comment:
>lookupResult IsNot Nothing AndAlso lookupResult.IsGood [](start = 39, length = 54)

Perhaps outside the scope of this PR, but the second half of this if exactly matches the next if. We should just combine them and nest the first part.

Reply:
Actually, wait, nevermind. I missed that this could change the result.

---
In reply to: [184231979](https://github.com/dotnet/roslyn/pull/26398#discussion_r184231979) [](ancestors = 184231979)